src/frontend/components/component-info/component-info.ts

```diff
@@ -32,68 +32,47 @@ export class ComponentInfo {
 
   private ComponentLoadState = ComponentLoadState;
 
-  private get path(): Path {
-    return deserializePath(this.node.id);
-  }
-
-  private get inputs(): InputOutput {
-    if (this.node == null || this.node.input == null) {
-      return {};
-    }
-
-    return this.node.input.reduce(
-      (accum, input) => {
-        const [name, alias] = input.split(/:/);
-        accum[name] = {alias};
-        return accum;
-      },
-      <InputOutput> {});
-  }
-
-  private get outputs(): InputOutput {
-    if (this.node == null) {
-      return {};
-    }
+  path: Path;
 
-    return this.node.output.reduce(
-      (accum, output) => {
-        const [name, alias] = output.split(/:/);
-        accum[name] = {alias};
-        return accum;
-      },
-      <InputOutput> {});
-  }
+  hasProviders: boolean = false;
+  hasDirectives: boolean = false;
+  hasDependencies: boolean = false;
+  hasProperties: boolean = false;
+  hasState: boolean = false;
 
-  private get hasState() {
-    if (this.node == null || this.state == null) {
-      return false;
-    }
+  inputs: InputOutput;
+  outputs: InputOutput;
 
-    return Object.keys(this.state).length > 0;
+  ngOnInit() {
+    this.ngOnChanges();
   }
 
-  private get hasProviders() {
-    return this.node &&
-        this.node.providers &&
-        this.node.providers.length > 0;
-  }
+  ngOnChanges() {
```

Comment:
OnChanges includes changing the node, which changes essentially all of these properties anyway.

Also note that previously these were getters, some of them referenced several times in the template, which means that (although they don't change) the values were rebuilt several times per template render.